### PR TITLE
"Demo" container with custom genesis

### PR DESCRIPTION
**Pull request by emberian**
This adds a new `coda-demo` container to our suite of services. This container uses the regular release deb, but builds its own genesis ledger with a single account containing all the stake. I've repurposed `docs/demo.md` to document this, the old demo that it documents is no longer maintained now that we have testnets.

### CODE REVIEW

File: dockerfiles/Dockerfile-coda-demo [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# coda package
+RUN echo "deb [trusted=yes] http://packages.o1test.net $deb_repo main" > /etc/apt/sources.list.d/coda.list \
+  && apt-get update \
+  && apt-get install --force-yes coda-testnet-postake-medium-curves=$coda_version -y \
+  && mkdir /root/keys \
+  && chmod go-rwx /root/keys \
+  && echo '{"box_primitive":"xsalsa20poly1305","pw_primitive":"argon2i","nonce":"7S1YA5PinXhnLgLJ3xemVnVPWdJdhKZ9RSNQbns","pwsalt":"AzDoECCYyJL8KuoB2vrsVc9Wg3xJ","pwdiff":[134217728,6],"ciphertext":"5UQuiQVbXPmR63ikri792dWR6Dz5dYZm8dLzwDyqWovdP5CzrLY6Fjw3QTHXA9J3PDkPZpvhrQfGkgU81kr9184dfoJDhn5EXxJMCAM44SZdmBYVszEQaSQnyy4BwsbRXmfjBMSW9ooGu2a5dFi5KHX5na6fr62VUB"}' > /root/keys/demo-block-producer \
+  && chmod go-rwx /root/keys/demo-block-producer \
+  && rm /var/lib/coda/coda_genesis_* \
+  && echo '{"pk":"4vsRCVMNTrCx4NpN6kKTkFKLcFN4vXUP5RB9PqSZe1qsyDs4AW5XeNgAf16WUPRBCakaPiXcxjp6JUpGNQ6fdU977x5LntvxrSg11xrmK6ZDaGSMEGj12dkeEpyKcEpkzcKwYWZ2Yf2vpwQP","balance":100000,"sk":null,"delegate":null}' >> /genesis-accounts.json \
```
Comment (bkase) [inline dockerfiles/Dockerfile-coda-demo]: Should this be `> /genesis-accounts.json`? Not that it should make a difference in this case

File: scripts/run-demo.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+set -e
+
+genesis_time=$(date -d "$(coda advanced compile-time-constants | jq -r '.genesis_state_timestamp')" +%s)
+now_time=$(date +%s)
+
+export CODA_TIME_OFFSET=$(( $now_time - $genesis_time ))
+export CODA_PRIVKEY_PASS=""
+
+exec coda daemon -block-producer-key /root/keys/demo-block-producer -run-snark-worker 4vsRCVMNTrCx4NpN6kKTkFKLcFN4vXUP5RB9PqSZe1qsyDs4AW5XeNgAf16WUPRBCakaPiXcxjp6JUpGNQ6fdU977x5LntvxrSg11xrmK6ZDaGSMEGj12dkeEpyKcEpkzcKwYWZ2Yf2vpwQP $@
```
Comment (Schmavery) [inline scripts/run-demo.sh]: We need to add `-insecure-rest-server` to be able to use the graphql port externally

File: scripts/run-demo.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+set -e
+
+genesis_time=$(date -d "$(coda advanced compile-time-constants | jq -r '.genesis_state_timestamp')" +%s)
+now_time=$(date +%s)
+
+export CODA_TIME_OFFSET=$(( $now_time - $genesis_time ))
+export CODA_PRIVKEY_PASS=""
+
```
Comment (Schmavery) [inline scripts/run-demo.sh]: Can we add something like
```
mkdir -p ~/.coda-config/wallets/store
echo '4vsRCVMNTrCx4NpN6kKTkFKLcFN4vXUP5RB9PqSZe1qsyDs4AW5XeNgAf16WUPRBCakaPiXcxjp6JUpGNQ6fdU977x5LntvxrSg11xrmK6ZDaGSMEGj12dkeEpyKcEpkzcKwYWZ2Yf2vpwQP' >  ~/.coda-config/wallets/store/4vsRCVMNTrCx4NpN6kKTkFKLcFN4vXUP5RB9PqSZe1qsyDs4AW5XeNgAf16WUPRBCakaPiXcxjp6JUpGNQ6fdU977x5LntvxrSg11xrmK6ZDaGSMEGj12dkeEpyKcEpkzcKwYWZ2Yf2vpwQP.pub
cp  /root/keys/demo-block-producer ~/.coda-config/wallets/store/4vsRCVMNTrCx4NpN6kKTkFKLcFN4vXUP5RB9PqSZe1qsyDs4AW5XeNgAf16WUPRBCakaPiXcxjp6JUpGNQ6fdU977x5LntvxrSg11xrmK6ZDaGSMEGj12dkeEpyKcEpkzcKwYWZ2Yf2vpwQP
```
To act as a hacky import (I need to edit `coda accounts import` to support non-interactive use...)

File: docs/demo.md [hunk 4]
```diff
@@ -1,163 +1,19 @@
...
-
-#### Interacting with the CLI
-
-To interact with the CLI, go to the compute engine -> VM instances page.
-
-Then, select the VM, start the VM, and click the SSH button to open up a terminal window.
-
+The demo container will run a block producer and a snark worker. You need to
+make sure the `--publish` at least the GraphQL port, for example `docker run
+--publish 3085:3085 -it codaprotocol/coda-demo`. Any additional arguments will
```
Comment (Schmavery) [inline docs/demo.md]: FWIW when I run this, I get
```
$ docker run --publish 3085:3085 -it codaprotocol/coda-demo
Unable to find image 'codaprotocol/coda-demo:latest' locally
docker: Error response from daemon: manifest for codaprotocol/coda-demo:latest not found: manifest unknown: manifest unknown.
```

works when I put something like codaprotocol/coda-demo:0.0.12-beta-fix-coda-demo-0308cf7 though

**Pull request merged**